adam/perception/perception_graph.py

```diff
@@ -355,6 +355,25 @@ def from_dynamic_perceptual_representation(
     ) -> "PerceptionGraph":
         return _FrameTranslation().translate_frames(perceptual_representation)
 
+    @staticmethod
+    def add_temporal_scopes_to_edges(
+        digraph: DiGraph, temporal_scopes: Union[TemporalScope, Iterable[TemporalScope]]
+    ) -> DiGraph:
+        """
+        Modifies the given digraph in place, applying the given `TemporalScope`\ s to all edges.
+        This new graph will be dynamic.
+
+        Note that this should only be applied to static perception digraphs.
```

Comment:
Can you add a check here for if the incoming dygraph is already marked and dynamic and if so throw a runtime error?

Reply:
@LichMaster98 Good thought. I added the check. It samples and checks an arbitrary edge (the first one) for dynamicness. Let me know if there's a better way to do this.

Reply:
hm, only other way I could think of would take more time which given this process is designed to save time I think this quick method is probably sufficient. 

This code will probably get slightly modified when/if we move to MultiDiGraphs so we can re-evaluate there if needed.